libparsec/crates/platform_storage/src/web/certificates.rs

```diff
@@ -48,11 +179,121 @@ impl<'a> PlatformCertificatesStorageForUpdateGuard<'a> {
         timestamp: DateTime,
         encrypted: Vec<u8>,
     ) -> anyhow::Result<()> {
-        todo!()
+        Certificate::insert(
+            &Certificate {
+                certificate_timestamp: timestamp.get_f64_with_us_precision(),
+                certificate: encrypted.into(),
+                certificate_type: certificate_type.into(),
+                filter1,
+                filter2,
+            },
+            &self.transaction,
+        )
+        .await
     }
 
     pub async fn get_last_timestamps(&mut self) -> anyhow::Result<PerTopicLastTimestamps> {
-        todo!()
+        let mut common_certifs = Vec::new();
+        let mut sequester_certifs = Vec::new();
+        let mut realm_certifs = Vec::new();
+        let mut shamir_recovery_certifs = Vec::new();
+
+        for ty in COMMON_CERTIFICATES {
+            common_certifs.extend(
+                &mut Certificate::get_values(
+                    &self.transaction,
+                    CertificateFilter {
+                        certificate_type: ty.into(),
+                        filter1: None,
+                        filter2: None,
+                    },
+                )
+                .await?
+                .into_iter()
+                .map(|certif| DateTime::from_f64_with_us_precision(certif.certificate_timestamp)),
+            );
+        }
+
+        let common_last_timestamp = common_certifs.into_iter().max_by(|x, y| x.cmp(y));
```

Comment:
Why the DateTime are stored in Vec, where we only use the greater one ?

Reply:
How do you prefer to find the greater one ?

Reply:
I was think of something like:

```suggestion
        let mut common_last_timestamp = None;
        
        for ty in COMMON_CERTIFICATES {
            common_last_timestamp = Certificate::get_values(
                    &self.transaction,
                    CertificateFilter {
                        certificate_type: ty.into(),
                        filter1: None,
                        filter2: None,
                    },
                )
                .await?
                .into_iter()
                .map(|certif| DateTime::from_f64_with_us_precision(certif.certificate_timestamp))
                .chain(common_last_timestamp)
                .max_by(|x, y| x.cmp(y));
        }
```

That convert the value to `DateTime` like the current solution but it chain the previous `common_last_timestamp`, then the resulting iterator is used with `max_by`

Reply:
Oh ok :+1: yes it is better